vignettes/test-children.Rmd

```diff
@@ -0,0 +1,181 @@
+---
+title: "Testing child packages"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Testing child packages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette is written for authors of packages that depend on crunch, who want
+to use similar testing infrastructure as the main crunch package. The crunch
+package itself has a long, evolutionary history, so not all of this advice applies 
+directly to developing for rcrunch itself 
+([more details at bottom of this document](#rcrunch-specific)). 
+
+When testing code that relies on an API, there are two broad categories of testing:
+
+- **Integration testing** - The tests call to the server each time they are run. 
+  While this is nice in theory, it can cause problems because test code can
+  overload the servers because of how frequently they need to run which will not only
+  cause your tests to fail, but negatively affect all users of the API.
+- **Mock testing** - Rather than actually call the server, the tests use "mocks"
+  of what should be returned by the server. This allows tests to be run more 
+  quickly and without stressing the server. The hard part is creating these mocks,
+  but rcrunch has tools that try to make this as easy as possible.
+
+
+## Integration testing strategy
+We do not advise that child packages run integration tests. If you feel you need them, 
+please contact the Crunch team and we may be able to grant you access to the 
+Crunch.io test server and help you write tests that use that.
+
+
+## Mock testing strategy
+Mock tests for the crunch package are tested using the `httptest` package, which was 
+written while the author worked at Crunch.io and so is tailored to our situation quite
+well. Before continuing, I highly recommend reading the vignettes for that package:
+
+- [`httptest`: A Test Environment for HTTP Requests in R](https://enpiar.com/r/httptest/articles/httptest.html)
+- [Redacting and Modifying Recorded Requests](https://enpiar.com/r/httptest/articles/redacting.html)
+
+Another strategy for generating mock tests is to use the 
```

Comment:
@1beb - added a short description of `with_mock()` here. In an ideal world I think you'd use `httptest` because it gives rcrunch more flexibility, though I understand it may not feel like it is worth the hassle to you.

Reply:
I think the suggestion that httptest is any better than with_mock is inappropriate. In both cases a real change to the API means you have to update your tests/code/fixture. Both are equally divorced :safety_pin: